webflow-hacks/form-hacks/18/18.ts

```diff
@@ -0,0 +1,40 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const invalidDomains = ['gmail.com', 'yahoo.com', 'hotmail.com', 'competitor.com'];
+  const FORM_SELECTOR = '[fs-hacks-element="form"]';
+  const EMAIL_SELECTOR = '[fs-hacks-element="email"]';
+  const formElement = document.querySelector<HTMLFormElement>(FORM_SELECTOR);
+  const emailInput = document.querySelector<HTMLInputElement>(EMAIL_SELECTOR);
+  if (!formElement || !emailInput) return;
+
+  formElement.addEventListener('submit', function (e) {
+    if (!validateEmail(emailInput.value)) {
+      emailInput.value = '';
+      emailInput.setAttribute('placeholder', 'Please enter a valid email address');
+      e.stopPropagation();
+      e.preventDefault();
+    }
+
+    const domainPart = emailInput.value.split('@');
+    if (!domainPart) {
+      e.stopPropagation();
+      e.preventDefault();
+    } else if (invalidDomains.includes(domainPart[1])) {
+      emailInput.value = '';
+      emailInput.setAttribute('placeholder', 'Please enter a business email');
+      e.stopPropagation();
+      e.preventDefault();
+    }
+  });
```

Comment:
I think all this domain check logic should be included inside the `validateEmail` function.
That function should check if the email is valid and if the domain is allowed, and return true or false + the correct message to display in case of error.